Java/core/src/main/java/com/amazon/randomcutforest/RandomCutForest.java

```diff
@@ -154,9 +169,21 @@
      */
     protected final boolean storeSequenceIndexesEnabled;
     /**
-     * Store the time information
+     * Enable compact representation
      */
     protected final boolean compactEnabled;
+    /**
+     * Enable float representation; the input/output is double[] but the internal
+     * representation would be float[]. There will be loss of precision in near
+     * neighbors because of the truncation. Otherwise, unless all numbers are in a
+     * very small range, the loss of precision is unlikely to have larger impact
+     * than changing the random seed.
+     */
+    protected final boolean floatEnabled;
+    /**
+     * The following set to false saves space, but enforces compact representation.
+     */
+    protected final boolean cacheEnabled;
```

Comment:
Maybe this variable name should indicate that it's specifically for caching or saving bounding boxes.

Also, are we going to throw an exception if compact enabled is false and cache enabled is true, or will we just silently ignore the value of compact enabled?

Reply:
Name changed to boundingBoxCachingEnable and is true by default (original behavior). In this PR we silently ignore compactEnabled in the builder  if some other feature mandates the compact representation. We check the consistency at the forest level.